requestor/ext/requestor-gwtjackson/src/main/java/io/reinert/requestor/ext/gwtjackson/SerializationModuleGenerator.java

```diff
@@ -0,0 +1,187 @@
+
+        moduleTypeElement = (TypeElement) element;
+
+        checkInheritsFromSerializationModule(moduleTypeElement);
+
+        moduleTypeInfo = new TypeInfo(moduleTypeElement.getQualifiedName().toString());
+
+        serdesAssemblers = new LinkedHashSet<JsonObjectSerdesAssembler>();
+
+        processAnnotationValues(moduleTypeElement);
+
+        serializationModuleAssembler = new SerializationModuleAssembler(moduleTypeInfo, serdesAssemblers);
+    }
+
+    public Set<JsonObjectSerdesAssembler> getSerdesAssemblers() {
+        return serdesAssemblers;
+    }
+
+    public boolean isGenerated() {
+        return generated;
+    }
+
+    public void generate(Map<TypeInfo, JsonObjectSerdesGenerator> serdesGenerators, Filer filer) throws ProcessingException {
+        try {
+            mergeAndAssembleSerdes(serdesGenerators, filer);
+            serializationModuleAssembler.assemble();
+            JavaFile.builder(serializationModuleAssembler.packageName(), serializationModuleAssembler.spec())
+                    .build().writeTo(filer);
+        } catch (Exception e) {
+            throw new ProcessingException(moduleTypeElement, e,
+                    "Error while writing generated code of %s: %s", moduleTypeInfo.getSimpleName(), e.getMessage());
+        }
+        generated = true;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o)
+            return true;
+        if (!(o instanceof SerializationModuleGenerator))
+            return false;
+
+        final SerializationModuleGenerator that = (SerializationModuleGenerator) o;
+
+        return moduleTypeInfo.equals(that.moduleTypeInfo);
+    }
+
+    @Override
+    public int hashCode() {
+        return moduleTypeInfo.hashCode();
+    }
+
+    private void checkElementIsInterface(Element element) throws ProcessingException {
+        if (element.getKind() != ElementKind.INTERFACE) {
+            throw new ProcessingException(element, "Only interfaces can be annotated with @%s",
+                    JsonSerializationModule.class.getSimpleName());
+        }
+    }
+
+    private void checkInheritsFromSerializationModule(TypeElement element) throws ProcessingException {
+        for (TypeMirror superItf : element.getInterfaces()) {
+            if (SerializationModule.class.getName().equals(superItf.toString()))
+                return; // is valid
+        }
+
+        throw new ProcessingException(element, "The interface %s annotated with @%s must inherit directly from %s",
+                element.getQualifiedName().toString(),
+                JsonSerializationModule.class.getSimpleName(),
+                SerializationModule.class.getName());
+    }
+
+    private void mergeAndAssembleSerdes(Map<TypeInfo, JsonObjectSerdesGenerator> externalGenerators, Filer filer)
+            throws ProcessingException {
+        for (JsonObjectSerdesAssembler assembler : new LinkedHashSet<JsonObjectSerdesAssembler>(serdesAssemblers)) {
```

Comment:
Ok. Imho however that 120 chars is a hard limit. I personally try to have shorter lines than that, so that we can always see the entire line in GitHub.